python/cudf/cudf/core/reshape.py

```diff
@@ -1160,3 +1160,260 @@ def _length_check_params(obj, columns, name):
+        - If margins is `True`, will also normalize margin values.
+
+    Returns
+    -------
+    DataFrame
+        Cross tabulation of the data.
+
+    Examples
+    --------
+    >>> a = np.array(["foo", "foo", "foo", "foo", "bar", "bar",
+    ...               "bar", "bar", "foo", "foo", "foo"], dtype=object)
+    >>> b = np.array(["one", "one", "one", "two", "one", "one",
+    ...               "one", "two", "two", "two", "one"], dtype=object)
+    >>> c = np.array(["dull", "dull", "shiny", "dull", "dull", "shiny",
+    ...               "shiny", "dull", "shiny", "shiny", "shiny"],
+    ...              dtype=object)
+    >>> cudf.crosstab(a, [b, c], rownames=['a'], colnames=['b', 'c'])
+    b   one        two
+    c   dull shiny dull shiny
+    a
+    bar    1     2    1     0
+    foo    2     2    1     2
+    """
+    if normalize is not None:
+        raise NotImplementedError("normalize is not supported yet")
+
+    if values is None and aggfunc is not None:
+        raise ValueError("aggfunc cannot be used without values.")
+
+    if values is not None and aggfunc is None:
+        raise ValueError("values cannot be used without an aggfunc.")
+
+    if not pd.core.dtypes.inference.is_nested_list_like(index):
+        index = [index]
+    if not pd.core.dtypes.inference.is_nested_list_like(columns):
+        columns = [columns]
+
+    rownames = (
+        _get_names(index, rownames, prefix="row") if not rownames else rownames
+    )
+    colnames = (
+        _get_names(columns, colnames, prefix="col")
+        if not colnames
+        else colnames
+    )
+
+    if len(index) != len(rownames):
+        raise ValueError("'index' and 'rownames' must have same length")
+    if len(columns) != len(colnames):
+        raise ValueError("'columns' and 'colnames' must have same length")
+
+    if len(set(rownames)) != len(rownames):
+        raise ValueError("rownames must be unique")
+    if len(set(colnames)) != len(colnames):
+        raise ValueError("colnames must be unique")
+
+    data = {
+        **dict(zip(rownames, map(as_column, index))),
+        **dict(zip(colnames, map(as_column, columns))),
+    }
+
+    df = cudf.DataFrame._from_data(data)
+
+    if values is None:
+        df["__dummy__"] = 0
+        kwargs = {"aggfunc": "count", "fill_value": 0}
+    else:
+        df["__dummy__"] = values
+        kwargs = {"aggfunc": aggfunc}
+
+    table = pivot_table(
+        data=df,
+        index=rownames,
+        columns=colnames,
+        values="__dummy__",
+        margins=margins,
+        margins_name=margins_name,
+        dropna=dropna,
+        **kwargs,
+    )
+
+    return table
+
+
+def pivot_table(
+    data,
+    values=None,
+    index=None,
+    columns=None,
+    aggfunc="mean",
+    fill_value=None,
+    margins=None,
+    dropna=None,
+    margins_name=None,
+    observed=None,
+    sort=None,
+):
+    """
+    Create a spreadsheet-style pivot table as a DataFrame.
```

Comment:
Can we expand the docstring to include a `Parameters`  and `Returns` section?

Reply:
Added the two sections.